papers/Kilpatrick2024_Alopeke/Tables/py/count_tables.py

```diff
@@ -0,0 +1,379 @@
+from astropy.io import fits
+from astropy.table import Table, vstack
+import sep
+import numpy as np
+import UTILS as ut
+import os
+import sys
+import glob
+import copy
+
+from astropy.time import Time
+from astropy import units as u
+from astropy.wcs import WCS, utils
+from astropy.coordinates import SkyCoord
+
+from astroquery.mast import Catalogs
+
+from photutils.detection import DAOStarFinder
+from astropy.stats import sigma_clipped_stats
+
+import progressbar
+
+def correct_masked_median_columns(fitsfile, camera):
+
+    image_data = np.median(fitsfile[0].data, axis=0)
+
+    for i in np.arange(fitsfile[0].header['NAXIS2']):
+        med = np.median(image_data[:,i])
+        fitsfile[0].data[:,:,i] = fitsfile[0].data[:,:,i]-med
+
+    return(fitsfile)
+
+
+def get_frame_alignment(fitsfile, coord, camera, catalog, save_stack=''):
+
+    image_data = np.median(fitsfile[0].data, axis=0)
+
+    newhdu = fits.PrimaryHDU()
+    newhdu.data = image_data
+
+    # Initial WCS guess
+    newhdu.header['CTYPE1']='RA---TAN'
+    newhdu.header['CTYPE2']='DEC--TAN'
+    newhdu.header['CRVAL1']=coord.ra.degree
+    newhdu.header['CRVAL2']=coord.dec.degree
+    if camera=='r':
```

Comment:
are these alignment corrections valid for both epochs? @charlie_kilpatrick

Reply:
seems the second epoch is treated differently below...

Reply:
All epochs are treated the same.  Instead of hard coding the WCS solution from the previous version, I do a full solve for the WCS solution using as many crossmatched Pan-STARRS sources as I can resolve in the image.  This is what the `utils.fit_wcs_from_points` call does.  The lines here are just the initial guess of the WCS solution, which is the same for both epochs but slightly different for each camera because of the fixed but known offset in WCS terms between them (mainly rotation, see: https://www.gemini.edu/instrumentation/alopeke-zorro/capability).